### PR TITLE
Bug 1222207 - Wait for l10n.ready when switching views

### PR DESCRIPTION
**Pull request by stasm**
https://bugzilla.mozilla.org/show_bug.cgi?id=1222207

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d55c21bc0ec87bc352c628b9cd8c5da0c1870dd0

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=dfa62c7bba6234c0f82c92b0832bc58fa10fe6bf

File: apps/music/views/album-detail/view.js [hunk 1]
```diff
@@ -34,8 +34,8 @@ AlbumDetailView.prototype.destroy = function() {
 };
 
 AlbumDetailView.prototype.render = function() {
-  View.prototype.render.call(this); // super();
-  this.list.setModel(this.songs);
+  return this.list.setModel(this.songs)
+    .then(View.prototype.render.bind(this)); // super();
```
Comment (justindarc) [inline apps/music/views/album-detail/view.js]: I wonder if we shouldn't just add an `onRenderDone()` method to the base `View` class instead? It seems odd to me to have a `super()` call occur async like this.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c10809f01f34b97e95eded6efd3e8311189e7709

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a4b96af31fa941ff9cfe1b69f028f8552e0f2695

**Pull request merged**